app/post.go

```diff
@@ -44,6 +44,29 @@ func CreatePostAsUser(post *model.Post) (*model.Post, *model.AppError) {
 			err.StatusCode = http.StatusBadRequest
 		}
 
+		if err.Id == "api.post.create_post.town_square_read_only" {
+			uchan := Srv.Store.User().Get(post.UserId)
```

Comment:
@hmhealey this is the code repeat I was talking about. But not sure if changing the signature of `CreatePost` would create problems.

Reply:
I think this is fine to have some duplication here. It would be a bit nicer to not get the user twice, but that will only happen when a user is trying to post in the Town Square when it's set to read only, so it should be pretty rare